yt/utilities/cython_fortran_utils.pyx

```diff
@@ -0,0 +1,88 @@
+cimport numpy as np
+import numpy as np
+import cython
+from libc.stdio cimport *
+import struct
+
+cdef INT_SIZE = sizeof(int)
+cdef DOUBLE_SIZE = sizeof(double)
+
+
+cdef class FortranFile:
+    def __init__(self, str fname):
+        self.cfile = fopen(fname.encode('utf-8'), 'r')
+
+    cpdef void skip(self, int n=1):
+        cdef int s1, s2, i
+
+        for i in range(n):
+            fread(&s1, INT_SIZE, 1, self.cfile)
+            fseek(self.cfile, s1, SEEK_CUR)
+            fread(&s2, INT_SIZE, 1, self.cfile)
+
+    cdef int get_size(self, str dtype):
+        if dtype == 'i':
+            return 4
+        elif dtype == 'd':
+            return 8
+        elif dtype == 'f':
+            return 4
+        else:
+            return struct.calcsize(dtype)
+
+    cpdef np.ndarray read_vector(self, str dtype):
+        cdef int s1, s2, size
+        cdef np.ndarray data
+
+        size = self.get_size(dtype)
+
+        fread(&s1, INT_SIZE, 1, self.cfile)
+        data = np.empty(s1 // size, dtype=dtype)
+        fread(<void *>data.data, size, s1 // size, self.cfile)
+        fread(&s2, INT_SIZE, 1, self.cfile)
+
+        return data
+
+    cpdef int read_int(self):
+        cdef int s1, s2, size=sizeof(int)
+        cdef int data
+
+        fread(&s1, INT_SIZE, 1, self.cfile)
+        fread(&data, size, s1 // size, self.cfile)
+        fread(&s2, INT_SIZE, 1, self.cfile)
+
+        return data
+
+    cpdef dict read_attrs(self, object attrs):
+        cdef str dtype
+        cdef int n
+        cdef dict data
+        cdef key
+        cdef np.ndarray tmp
```

Comment:
If it matters for performance it might be possible to make this a typed memoryview but you'd need to use [fused types](http://cython.readthedocs.io/en/latest/src/userguide/fusedtypes.html)